lib/segment/src/index/sparse_index/sparse_vector_index.rs

```diff
@@ -26,28 +35,150 @@ impl<TInvertedIndex: InvertedIndex> SparseVectorIndex<TInvertedIndex> {
+                QueryVector::Context(_) => {
+                    return Err(OperationError::ValidationError {
+                        description: "Cannot context query sparse vectors".to_string(),
+                    })
+                }
+            };
+            let sparse_vector: &SparseVector = vector.to_vec_ref().try_into()?;
+            let mut search_context = SearchContext::new(
+                sparse_vector.to_owned(),
+                top,
+                &self.inverted_index,
+                is_stopped,
+            );
+            let points = search_context.search(&condition);
+            result.push(points);
+        }
+
+        Ok(result)
     }
 }
 
-// TODO impl VectorIndex for SparseVectorIndex
+impl<TInvertedIndex: InvertedIndex> VectorIndex for SparseVectorIndex<TInvertedIndex> {
+    fn search(
+        &self,
+        vectors: &[&QueryVector],
+        filter: Option<&Filter>,
+        top: usize,
+        _params: Option<&SearchParams>,
+        is_stopped: &AtomicBool,
+    ) -> OperationResult<Vec<Vec<ScoredPointOffset>>> {
+        let id_tracker = self.id_tracker.borrow();
+        let vector_storage = self.vector_storage.borrow();
+        let deleted_point_bitslice = id_tracker.deleted_point_bitslice();
+        let deleted_vectors = vector_storage.deleted_vector_bitslice();
+        // filter for deleted points
+        let not_deleted_condition = |idx: PointOffsetType| -> bool {
+            !deleted_point_bitslice
+                .get(idx as usize)
+                .map(|x| *x)
+                .unwrap_or(false)
+                && !deleted_vectors
+                    .get(idx as usize)
+                    .map(|x| *x)
+                    .unwrap_or(false)
+        };
+        match filter {
+            Some(filter) => {
+                let payload_index = self.payload_index.borrow();
+                // TODO(sparse) check cardinality and fall back to payload index scan of low cardinality (needs benchmarks)
+                let filter_context = payload_index.filter_context(filter);
+                let matches_filter_condition =
+                    |idx: PointOffsetType| -> bool { filter_context.check(idx) };
+                self.search_sparse(vectors, top, is_stopped, true, |idx| {
+                    not_deleted_condition(idx) && matches_filter_condition(idx)
+                })
+            }
+            None => {
+                // query sparse index directly
+                self.search_sparse(vectors, top, is_stopped, false, not_deleted_condition)
+            }
+        }
+    }
+
+    fn build_index(&mut self, _stopped: &AtomicBool) -> OperationResult<()> {
+        let borrowed_vector_storage = self.vector_storage.borrow();
+        let borrowed_id_tracker = self.id_tracker.borrow();
+        let points_count = borrowed_vector_storage.total_vector_count();
+        let mut ram_index = InvertedIndexRam::empty();
+        for id in 0..points_count as PointOffsetType {
```

Comment:
I propose to use ready iterator with filter of deleted flags:
```
for id in  id_tracker.iter_ids_excluding(deleted_bitslice) {
```
where `deleted_bitslice` is deleted flags from vector_storage